> [!WARNING]
> You have got a presubmit warning. Please address it if possible.

```
./patchfile_test.py
vpython3 ./patchfile_test.py (1.87s) failed
EEEEEEEEEEEEEEEEEEEE
======================================================================
ERROR: test_apply_broken_patch_plaster_broken (__main__.PatchfilePlasterApplyTest.test_apply_broken_patch_plaster_broken)
Broken patch path + broken plaster → PLASTER_BROKEN.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 797, in test_apply_broken_patch_plaster_broken
    patchfile, patch_path = self._setup_broken_and_patchfile()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 764, in _setup_broken_and_patchfile
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_broken_patch_plaster_fixed (__main__.PatchfilePlasterApplyTest.test_apply_broken_patch_plaster_fixed)
Broken patch path + working plaster → PLASTER_FIXED.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 789, in test_apply_broken_patch_plaster_fixed
    patchfile, patch_path = self._setup_broken_and_patchfile()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 764, in _setup_broken_and_patchfile
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_conflict_plaster_broken (__main__.PatchfilePlasterApplyTest.test_apply_conflict_plaster_broken)
Conflict path + broken plaster → PLASTER_BROKEN.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 782, in test_apply_conflict_plaster_broken
    patchfile = self._setup_conflict_and_patchfile()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 738, in _setup_conflict_and_patchfile
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_conflict_plaster_fixed (__main__.PatchfilePlasterApplyTest.test_apply_conflict_plaster_fixed)
Conflict path + working plaster → PLASTER_FIXED.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 775, in test_apply_conflict_plaster_fixed
    patchfile = self._setup_conflict_and_patchfile()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 738, in _setup_conflict_and_patchfile
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_broken (__main__.PatchfileTest.test_apply_broken)
Tests the behavior when applying a broken patchfile.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 290, in test_apply_broken
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_clean (__main__.PatchfileTest.test_apply_clean)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 231, in test_apply_clean
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_conflict (__main__.PatchfileTest.test_apply_conflict)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 116, in test_apply_conflict
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_conflict_with_whitespace_error (__main__.PatchfileTest.test_apply_conflict_with_whitespace_error)
Tests the behavior when applying a patch with whitespace errors.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 177, in test_apply_conflict_with_whitespace_error
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_on_deleted (__main__.PatchfileTest.test_apply_on_deleted)
Tests the behavior when applying a patch to a deleted file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 330, in test_apply_on_deleted
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_apply_on_deleted_with_whitespace_error (__main__.PatchfileTest.test_apply_on_deleted_with_whitespace_error)
Tests DELETED status when stderr has whitespace warnings before the
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 374, in test_apply_on_deleted_with_whitespace_error
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_get_last_commit_for_source (__main__.PatchfileTest.test_get_last_commit_for_source)
Test get_last_commit_for_source method.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 484, in test_get_last_commit_for_source
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_get_repository_from_patch_name (__main__.PatchfileTest.test_get_repository_from_patch_name)
Test the result of get_repository_from_patch_name
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 40, in test_get_repository_from_patch_name
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'build/android/gyp/dex.py']' returned non-zero exit status 255.

======================================================================
ERROR: test_get_source_removal_status (__main__.PatchfileTest.test_get_source_removal_status)
Test get_source_removal_status for a deleted source file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 548, in test_get_source_removal_status
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_get_source_rename_status (__main__.PatchfileTest.test_get_source_rename_status)
Test get_source_removal_status for a renamed source file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 605, in test_get_source_rename_status
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/common/extensions/api/developer_private.idl']' returned non-zero exit status 255.

======================================================================
ERROR: test_path_from_repo (__main__.PatchfileTest.test_path_from_repo)
Tests the path_from_repo method of Patchfile.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 450, in test_path_from_repo
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'build/android/gyp/dex.py']' returned non-zero exit status 255.

======================================================================
ERROR: test_plaster_none_for_non_chromium_repo (__main__.PatchfileTest.test_plaster_none_for_non_chromium_repo)
Plaster is None for patches targeting a sub-repository.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 680, in test_plaster_none_for_non_chromium_repo
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'v8']' returned non-zero exit status 255.

======================================================================
ERROR: test_plaster_none_when_file_missing (__main__.PatchfileTest.test_plaster_none_when_file_missing)
Plaster is None for a Chromium patch when no toml file exists.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 665, in test_plaster_none_when_file_missing
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_plaster_set_when_file_exists (__main__.PatchfileTest.test_plaster_set_when_file_exists)
Plaster path is set for a Chromium patch when the toml exists.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 643, in test_plaster_set_when_file_exists
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'chrome/browser/foo.cc']' returned non-zero exit status 255.

======================================================================
ERROR: test_source_from_brave (__main__.PatchfileTest.test_source_from_brave)
Tests the source_from_brave method of Patchfile.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 420, in test_source_from_brave
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'build/android/gyp/dex.py']' returned non-zero exit status 255.

======================================================================
ERROR: test_source_name_from_patch_naming (__main__.PatchfileTest.test_source_name_from_patch_naming)
Test patched file name name heuristics.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/./patchfile_test.py", line 70, in test_source_name_from_patch_naming
    self.fake_chromium_src.run_update_patches()
  File "/home/ubuntu/teamcity/work/space/src/brave/tools/cr/test/fake_chromium_repo.py", line 292, in run_update_patches
    result = subprocess.run(
             ^^^^^^^^^^^^^^^
  File "/home/ubuntu/teamcity/work/space/cache/vpython/store/cpython+l5fnajrvijf7cvdkjqmbicg3i8/contents/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'diff', '--src-prefix=a/', '--dst-prefix=b/', '--default-prefix', '--full-index', 'build/android/gyp/dex.py']' returned non-zero exit status 255.

----------------------------------------------------------------------
Ran 20 tests in 1.638s

FAILED (errors=20)

```

<!-- presubmit-hash=130096ab4efb217e3a5cce9fef0a02e4b05747fa9015f3368948e8ab126733a9 -->

--- a/tools/cr/toolchain/README.md
+++ b/tools/cr/toolchain/README.md
@@ -1,0 +1,38 @@
+# Browser Toolchain Scripts
+
+This directory contains scripts for generating browser toolchains (compilers,
+linkers, standard libraries, etc.) used by the Brave build.
+
+## Design rules
+
+Scripts here are designed to be launched with a single `curl` one-liner,
+so they must be **self-contained in a single source file** with **no
+dependencies outside the Python standard library** (or the platform's
+default tooling such as `git`). Do not split logic across helper modules or
+add third-party imports.
+
+Example invocation pattern:
+
+```sh
+curl -sL \
+    https://raw.githubusercontent.com/brave/brave-core/refs/heads/master/tools/cr/toolchain/<script>.py \
+    | python3 - [args...]
+```
+
+## Scripts
+
+### `build_rust_toolchain.py`
+
+Builds and packages a minimal Rust toolchain subset for Chromium: the
+`rust-lld` linker and the `wasm32-unknown-unknown` stage-1 standard-library
+sysroot. The output is a `.tar.xz` archive ready for upload to GCS.
+
+```sh
+curl -sL \
+    https://raw.githubusercontent.com/brave/brave-core/refs/heads/master/tools/cr/toolchain/build_rust_toolchain.py \
+    | python3 - \
+        --out-dir=./out/ \
+        --chromium-src=~/dev/chromium/src/
+```
+
+Pass `--help` for the full list of options.

--- a/tools/cr/toolchain/build_rust_toolchain.py
+++ b/tools/cr/toolchain/build_rust_toolchain.py
@@ -12,7 +12,7 @@ To use this straight from Github just call:
 
 ```sh
 curl -sL \
-    https://raw.githubusercontent.com/brave/brave-core/refs/heads/master/tools/rust/build_rust_toolchain_standalone.py \
+    https://raw.githubusercontent.com/brave/brave-core/refs/heads/master/tools/cr/toolchain/build_rust_toolchain.py \
     | python3 - \
         --out-dir=./out/ \
         --chromium-src=~/dev/chromium/src/
@@ -22,7 +22,7 @@ If you do not have a Chromium checkout yet, pass `--clone-chromium` together
 with `--use-ref` to have the script fetch one automatically:
 
 ```sh
-python3 build_rust_toolchain_standalone.py \
+python3 build_rust_toolchain.py \
     --out-dir=./out/ \
     --chromium-src=~/dev/chromium/src/ \
     --clone-chromium \
@@ -111,6 +111,7 @@ if sys.platform == 'win32':
     # Path to Git's sh.exe on Windows, which is used by
     # `tools/rust/build_rust.py` to build the toolchain on Windows.`
     GIT_SH_PRESUMED_BIN_PATH = Path(r'C:\Program Files\Git\bin\sh.exe')
+
 
 def _check_call(*command, cwd=None):
     """Run *command* as a subprocess, logging the invocation and any stderr.
@@ -208,7 +209,6 @@ class ToolchainBuilder:
 
         # Module for tools/rust/package_rust.py. Initialised by `run()`.
         self.package_rust_module: Optional[ModuleType] = None
-
 
     @contextlib.contextmanager
     def _temporary_config_toml_template_edits(self):

